services/publishing/Tweek.Publishing.Service/Sync/StorageSynchronizer.cs

```diff
@@ -64,31 +59,10 @@ public StorageSynchronizer(IObjectStorage storageClient, ShellHelper.ShellExecut
                 Previous = versionsBlob?.Latest,
             };
 
-            var (p, exited) = _shellExecutor("git", $"archive --format=zip {commitId}");
-            using (var ms = new MemoryStream())
+            foreach (var uploader in Uploaders)
             {
-                await p.StandardOutput.BaseStream.CopyToAsync(ms);
-                ms.Position = 0;
-                await exited;
+                await uploader.Upload(commitId);
```

Comment:
We should consider paralleling this in the future.

Reply:
issue added #776